### ISSUE TITLE
Binary Project Specification for Carthage

### ISSUE THREAD

**Issue by jimmyhoran** [Appboy/appboy-segment-ios#38]
After trying to use the new Carthage support, it wasn't until I tried to checkout and build the dependency with Carthage that I realised it doesn't support building with Carthage.

```shell
*** Skipped building appboy-ios-sdk due to the error:
Dependency "appboy-ios-sdk" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/appboy/appboy-ios-sdk/issues/new
*** Skipped building appboy-segment-ios due to the error:
Dependency "appboy-segment-ios" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/appboy/appboy-segment-ios/issues/new
```

It would be great if you could provide and manage a Binary Project Specification and consider updating the read me to mention that the dependency binary download only.
https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#binary-project-specification

**Comment by Bucimis** [comment]
Hi @jimmyhoran, 

1) Could you share the contents of your Cartfile for analytics-ios, appboy-ios-sdk and appboy-segment-ios? 
2) Could you try deleting your `Carthage` build folder and trying again?
3) Do you have Cocoapods installed on your machine, and if so, what version?

**Comment by jimmyhoran** [comment]
@Bucimis just to clarify for you, the error below is expected when running `carthage update --no-use-binaries`.
```shell
*** Skipped building appboy-ios-sdk due to the error:
Dependency "appboy-ios-sdk" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/appboy/appboy-ios-sdk/issues/new
*** Skipped building appboy-segment-ios due to the error:
Dependency "appboy-segment-ios" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/appboy/appboy-segment-ios/issues/new
```

Because `appboy-ios-sdk` and `appboy-segment-ios` dependencies are only available as binary,  I asking if you could publish a Binary Project Specification to make it easier to use the dependencies when using `carthage update --no-use-binaries`.
https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#binary-project-specification

At the moment we're having to host our own Binary Project Specification in our project to be able to use `appboy-ios-sdk` and `appboy-segment-ios`.

**Comment by Bucimis** [comment]
@jimmyhoran thanks for the additional information. `appboy-ios-sdk` already contains a Binary Project Specification (see https://github.com/Appboy/appboy-ios-sdk/blob/master/appboy_ios_sdk.json) and I was able to repro it not building with `--no-use-binaries` enabled. My guess is that `--no-use-binaries` doesn't respect Binary Project Specifications. What happens if you run `carthage update` without the `--no-use-binaries` flag?

**Comment by jimmyhoran** [comment]
> @jimmyhoran thanks for the additional information. `appboy-ios-sdk` already contains a Binary Project Specification (see https://github.com/Appboy/appboy-ios-sdk/blob/master/appboy_ios_sdk.json) and I was able to repro it not building with `--no-use-binaries` enabled. My guess is that `--no-use-binaries` doesn't respect Binary Project Specifications. What happens if you run `carthage update` without the `--no-use-binaries` flag?

@Bucimis that's is correct and expected, `--no-use-binary` requires an explicit reference to a binary specification like so:
```shell
github "segmentio/analytics-ios"
binary "appboy-segment-ios.binary.json" == 3.2.0
binary "https://github.com/Appboy/appboy-ios-sdk/blob/master/appboy_ios_sdk.json" == 3.24.1
```

Great to see that you have a binary project specification for `app-ios-sdk`, wasn't aware that was available.
Do you have a binary project specification available for `appboy-segment-ios`?

**Comment by Bucimis** [comment]
@jimmyhoran thanks for the breakdown, we'll add a binary project spec in our next release.

**Comment by jimmyhoran** [comment]
Awesome. Thanks @Bucimis  🙏

**Comment by chshapiro** [comment]
Hey @jimmyhoran,

This was released in version 3.3.0. Feel free to open a new issue if you have any further questions.
